### DISCUSSION TITLE
Compared with other coupling software, what are the characteristics or advantages of ARMI?

### DISCUSSION THREAD

**Discussion by Scienkiller** [terrapower/armi#296] [Q&A] [answered]
I have always wanted to make a multi-physical coupling software in the nuclear field. Now I am looking for a platform, and ARMI is a very good software. However, there is still a problem that it is not clear what the characteristics of ARMI are. Compared with other coupling software, such as SALOME, what are the features and advantages of ARMI?

**Comment by ntouran** [comment, marked as answer]
It'd be worth coming up with a more rigorous comparison matrix for the [various related tools](https://github.com/paulromano/awesome-nuclear). Barring that I'd say that ARMI's advantages are:

* It's totally focused on analysis of nuclear reactors, and so the [basic examples](https://terrapower.github.io/armi/gallery/index.html) and [tutorials](https://terrapower.github.io/armi/tutorials/index.html#) are directly relevant to typical nuclear engineering/design needs. Some more general frameworks are necessarily more generic and take more time to set up something nuclear-relevant. 
* As a corollary to above: ARMI provides a bunch of nuclear specialty tooling in the box. 
* It's all in Python so it's relatively easy for typical nuclear or mechanical engineers without C++ or Fortran backgrounds to get up and running quickly in ARMI. This is a blessing in that you can get a lot of sophisticated analysis automated rapidly, but also a curse in that you have to keep an eye on the code quality of the various plugins that they produce.
